src/silvistat/storage.py

```diff
@@ -1,68 +1,96 @@
 import pdal
 import tiledb
 import numpy as np
+from math import floor
 
 from . import Bounds
 
 class Storage(object):
-
-    def __init__(self, dirname, filename, config=None, cell_size=30,
+    """ Handles storage of shattered data in a TileDB structure. """
+    def __init__(self, dirname, filename, config= None, cell_size=30,
                  chunk_size=16):
         self.dirname = dirname
         self.src_file = filename
         self.config = config
         self.cell_size = cell_size
         self.chunk_size = chunk_size
+        ## TODO insert config here
+        self.ctx = tiledb.default_ctx()
 
+        self.atts = None
         self.schema = None
         self.tdb = None
         self.bounds = None
 
     def __inspect_file(self):
-        r = pdal.Reader(self.filename)
+        """
+        Does pdal quick info on source file for making informed decisions on
+        storage, adding available attributes and bounds if not supplied by user.
+        """
+        r = pdal.Reader(self.src_file)
         info = r.pipeline().quickinfo[r.type]
-        self.atts=info['dimensions']
+        if not self.atts:
+            self.atts=info['dimensions']
         if not self.bounds:
             b = info['bounds']
-            self.bounds = Bounds(b.minx, b.miny, b.maxx, b.maxy, self.cell_size,
-                                  self.chunk_size)
+            srs = info['srs']['wkt']
+            self.bounds = Bounds(b['minx'], b['miny'], b['maxx'], b['maxy'], self.cell_size,
```

Comment:
This bounds is a property of the database at initialize time. it shouldn't be plucked off a single file.

Reply:
Uncertain how we would create the database otherwise though. We need information to base the database creation off of. Same with the attributes portion. I have to know what attributes to add to the database at creation time